### PR TITLE
TE5-T605 Web Services Implemented Without 3rd party Library

### PR DESCRIPTION
**Pull request by jay-a-shah**

### CODE REVIEW

File: app/src/main/java/com/example/demoappkotlin/WebServices/ApiInterface.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package com.example.demoappkotlin.WebServices
+
+
```
Comment (Purvesh-Dodiya) [inline app/src/main/java/com/example/demoappkotlin/WebServices/ApiInterface.kt]: extra empty line

File: app/src/main/java/com/example/demoappkotlin/WebServices/ApiInterface.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            }
+        }
+
+        fun getRetrofitObject(): ApiInterface{
+            val retrofit =
+                Retrofit.Builder().baseUrl(BaseURL).addConverterFactory(GsonConverterFactory.create())
+                    .client(okHttpClientBuilder.build()).build().create(ApiInterface::class.java)
+            return retrofit
+        }
+
```
Comment (Purvesh-Dodiya) [inline app/src/main/java/com/example/demoappkotlin/WebServices/ApiInterface.kt]: extra empty line

**Review by Purvesh-Dodiya — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/demoappkotlin/WebServices/ApiInterface.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.POST
+import retrofit2.http.Path
+import com.example.demoappkotlin.BuildConfig
+
+interface ApiInterface {
+    @GET("/api/users/{id}")
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/WebServices/ApiInterface.kt]: move end point to constant

File: app/src/main/java/com/example/demoappkotlin/WebServices/LoginScreenWithApi.kt [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                catch (ex: Exception) {
+                    ex.printStackTrace()
+                } }.start()
+            showDialog()
+        }
+    }
+
+    private fun sendPostRequest(username: String, password: String) {
+        val credential = JSONObject()
+        credential.put("email", username)
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/WebServices/LoginScreenWithApi.kt]: move static key

File: app/src/main/java/com/example/demoappkotlin/WebServices/LoginScreenWithApi.kt [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        with(url.openConnection() as HttpURLConnection) {
+            requestMethod = "POST"
+            setRequestProperty("Content-Type", "application/json")
+            val wr = OutputStreamWriter(outputStream)
+            wr.write(credential.toString())
+            wr.flush()
+            println("URL : $url")
+            println("cred: $credential")
+            println("Response Code: $responseCode")
+            if (responseCode == 200) {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/WebServices/LoginScreenWithApi.kt]: move static key

**Review by purvesh-d — COMMENTED** [review]
Check for static key in whole file

**Review by purvesh-d — APPROVED** [review]

**Pull request merged**